frontend/src/pages/calculatedResults/index.js

```diff
@@ -1,13 +1,62 @@
 import React from 'react';
-import { PlaceholderStyle } from '../../components/placeholderStyle/PlaceholderStyleElement';
-import './CalculatedResultsElements';
+import { useLocation } from 'react-router-dom';
+import Footer from '../../components/footerPurple';
+import {
+  FixedInstallmentsContainer,
+  InstallmentContainer,
+  IllustrationContainer,
+  DescendingInstallmentsContainer,
+  YearTabs,
+} from './CalculatedResultsElements';
+import { SectionTitle } from '../../components/sectionTitle/SectionTitle';
+import { GradientText } from '../../components/gradientText/GradientText';
+import landingPageIllustration from '../../images/landingPageIllustration.svg';
+
+const { TabPane } = YearTabs;
 
 function CalculatedResults() {
+  const location = useLocation();
+
+  if (location.state.response.monthly_payment.length !== undefined) {
```

Comment:
Is `location.state.response.monthly_payment` always defined? Maybe it would be safer to use optional chaining here:
```
if (location.state.response?.monthly_payment?.length !== undefined) {
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
done